spyder/preferences/configdialog.py

```diff
@@ -449,28 +455,32 @@ def save_to_conf(self):
                     data = [item.strip() for item in data.split(',')]
                 else:
                     data = to_text_string(data)
-                self.set_option(option, data)
-        for spinbox, (option, _default) in list(self.spinboxes.items()):
+                self.set_option(option, data, section=sec)
+        for spinbox, (sec, option, _default) in list(self.spinboxes.items()):
             if option in self.changed_options:
-                self.set_option(option, spinbox.value())
-        for combobox, (option, _default) in list(self.comboboxes.items()):
+                self.set_option(option, spinbox.value(), section=sec)
+        for combobox, (sec, option, _default) in list(self.comboboxes.items()):
             if option in self.changed_options:
                 data = combobox.itemData(combobox.currentIndex())
-                self.set_option(option, from_qvariant(data, to_text_string))
+                self.set_option(option, from_qvariant(data, to_text_string),
+                                section=sec)
         for (fontbox, sizebox), option in list(self.fontboxes.items()):
             if option in self.changed_options:
                 font = fontbox.currentFont()
                 font.setPointSize(sizebox.value())
-                self.set_font(font, option)
-        for clayout, (option, _default) in list(self.coloredits.items()):
+                self.set_font(font, option, section=sec)

```

Comment:
@goanpeca I have a question.
Does set_font method have section keyword?
An error occurs when changing the font.